.github/workflows/codeql-python.yml

```diff
@@ -0,0 +1,42 @@
+# GitHub action to run static code analysis on C++ and Python code
+name: "CodeQL Code Analysis"
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['python']
+
+    steps:
+    - name: Checkout repository
+      if: ${{ github.repository == 'lammps/lammps' }}
```

Comment:
You could move this `if:` up a level, then it would apply to all steps and make the code more readable.
(see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif). Plus `github.repository_owner == 'lammps' might be simpler.